src/index.ts

```diff
@@ -134,11 +138,16 @@ function requestToFetchOptions(reqOpts: Options) {
     process.env.HTTP_PROXY ||
     process.env.http_proxy ||
     process.env.HTTPS_PROXY ||
-    process.env.https_proxy;
-  if (reqOpts.proxy || proxy) {
-    options.agent = new HttpsProxyAgent(proxy);
+    process.env.https_proxy ||
+    reqOpts.proxy;
```

Comment:
reqOpts.proxy should probably precede the environment variables, since it's possible the user wants to override the environment.